whois-update/src/main/java/net/ripe/db/whois/update/handler/validator/domain/NServerValidator.java

```diff
@@ -66,4 +72,19 @@ public ImmutableList<Action> getActions() {
     public ImmutableList<ObjectType> getTypes() {
         return TYPES;
     }
+
+
+    private void validateNserverCorrectPrefixes(RpslObject rpslObject, int prefixLength, boolean isIpv4){
+        if (hasRipeNserver(rpslObject) && hasIncorrectPrefixes(prefixLength, isIpv4)){
+            throw new BadRequestException();
+        }
+    }
+
+    private boolean hasIncorrectPrefixes(int prefixLength, boolean isIpv4) {
+        return prefixLength!=32 && !isIpv4 || prefixLength!=16 && isIpv4;
+    }
+
+    private boolean hasRipeNserver(RpslObject rpslObject) {
+        return rpslObject.findAttributes(AttributeType.NSERVER).stream().anyMatch(nserver -> nserver.getValue().equals("ns.ripe.net"));
+    }
```

Comment:
This needs to be case insensitive? Or CIString is handling this case already?

Reply:
need case sensitive